src/main/java/hudson/plugins/jdepend/JDependBuildAction.java

```diff
@@ -8,23 +8,35 @@
 
 import hudson.model.AbstractBuild;
 import hudson.model.Action;
+import hudson.model.Run;
+import jenkins.model.RunAction2;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+import javax.annotation.CheckForNull;
 
 /**
  * A build action to generate JDepend HTML reports
  * @author cflewis
  *
  */
-public class JDependBuildAction implements Action 
+public class JDependBuildAction implements RunAction2
 {
-	public final AbstractBuild<?, ?> build;
-	private final JDependParser jDependParser;
 	private String htmlReport;
-	
-	public JDependBuildAction(AbstractBuild<?, ?> build, JDependParser jDependParser)
+	private transient Run<?,?> owner;
+
+	public JDependBuildAction(JDependParser jDependParser) {
+		this(null, jDependParser);
+	}
+
+	/**
+	 * @deprecated Use {@link #JDependBuildAction(JDependParser)}
+	 */
+	@Deprecated
```

Comment:
Probably pointless if you are deleting the `build` field anyway. Who would be calling this method from outside anyway?

Reply:
🤷‍♂️  Prefer to keep binary compatibility when it does not cost too much